### ISSUE TITLE
docker too many open files due to fluentd buffer

### ISSUE THREAD

**Issue by jagjag** [rancher/rancher#18480]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
1.  project /  resources / project logging 
    add  elasticsearch  logging target 
2.  run you project and wait

**Result:**

![image](https://user-images.githubusercontent.com/1321754/53557247-79fd3200-3b80-11e9-82e1-924e6af7002d.png)

![image](https://user-images.githubusercontent.com/1321754/53557261-82556d00-3b80-11e9-9667-f6450b8a0c5f.png)

we  found  thant  the fluentd process opened too many buffer files

![image](https://user-images.githubusercontent.com/1321754/53557220-694cbc00-3b80-11e9-8b64-0c5754f4ed88.png)

**Other details that may be helpful:**
 We restarted docker and disabled fluentd logging , then problem solved.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.1.3
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): imported kubeadm 1.11.6
- Machine type (cloud/VM/metal) and specifications (CPU/memory):  8c / 16G
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.6", GitCommit:"b1d75deca493a24a2f87eb1efde1a569e52fc8d9", GitTreeState:"clean", BuildDate:"2018-12-16T04:39:52Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.6", GitCommit:"b1d75deca493a24a2f87eb1efde1a569e52fc8d9", GitTreeState:"clean", BuildDate:"2018-12-16T04:30:10Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
docker version
Client:
 Version:      17.03.3-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   e19b718
 Built:        Thu Aug 30 01:04:51 2018
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.3-ce
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   e19b718
 Built:        Thu Aug 30 01:04:51 2018
 OS/Arch:      linux/amd64
 Experimental: false
```

**Comment by aiwantaozi** [comment]
Hi, @jagjag We fixed it in 2.2, now have some release candidate version, you can try it, v2.2.0-rc3

**Comment by jagjag** [comment]
@aiwantaozi  thank you, I will try it.

**Comment by ibaldonl** [comment]
Any workarounds for 2.1 meanwhile?
Thanks!

**Comment by aiwantaozi** [comment]
fluentd 1.1.3 will create buffer file each flush interval and if the flush interval is very small, there are lots of small queued chunks in buffer. This is not good with file buffer because it consumes lots of fd resources when output destination doesn't have enough throughput. You can try increase the flush interval.

**Comment by loganhz** [comment]
Close as it has been fixed in 2.2
